torch/utils/data/dataset.py

```diff
@@ -291,6 +298,11 @@ def __init__(self, dataset: Dataset[T_co], indices: Sequence[int]) -> None:
         self.dataset = dataset
         self.indices = indices
 
+        # add batched sampling support when parent dataset supports it.
+        # see torch.utils.data._utils.fetch._MapDatasetFetcher
+        if getattr(dataset, "__getitems__", None):
```

Comment:
I think Python does have a proper way to do it:
```suggestion
        if hasattr(dataset, "__getitems__"):
```

Reply:
@ejguan This is a bit different semantically, the original would still return false if `a.__getitems__` = False etc. while hasattr would return true in that case.

Reply:
The original code was `hasattr(dataset, "__getitems__") and dataset.__getitems__` which caused mypy to complain unlike this version.

Reply:
Wait, what is the case that `__getitems__` becomes an attribute rather than a method? Then, why not `hasattr(dataset, "__getitems__") and callable(dataset.__getitems__)`

Reply:
@ejguan That's a double lookup on `__getitems__` better to do `callable(getattr(dataset, "__getitems__", None))` which is more efficient.

Reply:
Originally I kept the same behavior as  torch.utils.data._utils.fetch._MapDatasetFetcher
```
class _MapDatasetFetcher(_BaseDatasetFetcher):
    def fetch(self, possibly_batched_index):
        if self.auto_collation:
            if hasattr(self.dataset, "__getitems__") and self.dataset.__getitems__:
```

Reply:
Both call patterns should probably changed to be honest.

Reply:
Probably better to do the same for fetcher, but it's out of scope of this PR.